src/NUnitTestAdapterTests/TestDiscoveryTests.cs

```diff
@@ -31,6 +31,8 @@ public void LoadMockassembly()
             Assert.That(NUnit.Tests.Assemblies.MockAssembly.Tests, Is.EqualTo(31),
                 "The reference to mock-assembly.dll appears to be the wrong version");
 
+            TestCases = new List<TestCase>();
```

Comment:
Actually, it is. The test class implements ITestCaseDiscoverySink and receives events as each test is found. The tests are saved in the list for use in the test methods.